cmd/gateway/s3/gateway-s3.go

```diff
@@ -635,6 +656,50 @@ func (l *s3Objects) DeleteBucketPolicy(ctx context.Context, bucket string) error
 	return nil
 }
 
+// GetObjectTag gets the tags set on the object
+func (l *s3Objects) GetObjectTag(ctx context.Context, bucket string, object string) (tagging.Tagging, error) {
+	var err error
+	var tags *tagging.Tagging
+	var tagStr string
+	var opts minio.ObjectOptions
+
+	if _, err = l.GetObjectInfo(ctx, bucket, object, opts); err != nil {
+		return tagging.Tagging{}, minio.ErrorRespToObjectError(err, bucket, object)
+	}
+
+	if tagStr, err = l.Client.GetObjectTagging(bucket, object); err != nil {
+		return tagging.Tagging{}, minio.ErrorRespToObjectError(err, bucket, object)
+	}
+	if tags, err = tagging.ParseTagging(strings.NewReader(tagStr)); err != nil {
+		return tagging.Tagging{}, minio.ErrorRespToObjectError(err, bucket, object)
```

Comment:
what are the error scenarios here? Have you verified if gateway returns same errors in that scenario as AWS S3?